### PR TITLE
vt/mysqlctl: improve backup progress reporting (Fixes #13497)

### LINKED ISSUE #13497: Bug Report: backups progress logs are incorrect
### Overview of the Issue

During the backup restore phase when downloading/reading files the progress reporting is incorrect:

```text
builtinbackupengine.go:666] Done taking Backup "90"
```

This is misleading because vtbackup is [downloading a file](https://github.com/vitessio/vitess/blob/7e649f8466c6aa2cca7e13d5379a199f4e97f42e/go/vt/mysqlctl/builtinbackupengine.go#L713), also `90` is actually the [index](https://github.com/vitessio/vitess/blob/7e649f8466c6aa2cca7e13d5379a199f4e97f42e/go/vt/mysqlctl/builtinbackupengine.go#L971) of the file in the internal slice, ideally it should show the filename.

### Reproduction Steps

Take a backup using any backup engine.

### Binary Version

```sh
λ ~ vtbackup --version
Version: 18.0.0-SNAPSHOT (Git revision  branch '') built on  by @ using go1.20.5 darwin/arm64
```

### Operating System and Environment details

```sh
any OS
```

### Log Fragments

```sh
I0714 10:46:12.806968       1 builtinbackupengine.go:686] Done taking Backup "0"
I0714 10:46:12.886414       1 builtinbackupengine.go:940] Copying file 71: performance_schema/events_transacti_137.sdi
I0714 10:46:13.084544       1 builtinbackupengine.go:686] Done taking Backup "143"
I0714 10:46:13.280642       1 builtinbackupengine.go:940] Copying file 1: ibdata1
I0714 10:46:13.293631       1 compression.go:166] Decompressing using external command: "nice -n 19 zstd -c -d"
I0714 10:46:13.677064       1 compression.go:166] Decompressing using external command: "nice -n 19 zstd -c -d"
I0714 10:46:13.786139       1 builtinbackupengine.go:686] Done taking Backup "71"
I0714 10:46:13.887079       1 builtinbackupengine.go:940] Copying file 2: ib_logfile0
I0714 10:46:14.383994       1 compression.go:166] Decompressing using external command: "nice -n 19 zstd -c -d"
I0714 10:46:15.684373       1 builtinbackupengine.go:686] Done taking Backup "1"
I0714 10:46:15.692404       1 builtinbackupengine.go:940] Copying file 3: ib_logfile1
```

### PR DESCRIPTION
**Pull request by yields**
## Description

During restore the logs are incorrect, the output looks like this:

    Done taking backup: "<filename>"

The commit fixes the issue, during backup the logs will
look like this:

        backup: "<filename>"
        reading "<filename>": <size>kb
        writing "<filename>": <size>kb
        finished reading: "<filename>" (<size>kb)
        finished writing: "<filename>" (<size>kb)
        finished backup: "<filename>"

during restore:

        restore: "<filename>"
        reading "<filename>": <size>kb
        writing "<filename>": <size>kb
        finished reading: "<filename>" (<size>kb)
        finished writing: "<filename>" (<size>kb)
        finished restore: "<filename>"

## Related Issue(s)

Fixes https://github.com/vitessio/vitess/issues/13497

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/mysqlctl/builtinbackupengine.go [hunk 1]
```diff
@@ -694,15 +697,22 @@ func (bp *backupPipe) ReportProgress(period time.Duration, logger logutil.Logger
...
+			n := float64(atomic.LoadInt64(&bp.nn))
 			if bp.maxSize == 0 {
-				logger.Infof("Backup %q: %.02fkb", bp.filename, written/1024.0)
+				logger.Infof("%s %q: %.02fkb", bp.action, bp.filename, n/1024.0)
 			} else {
 				maxSize := float64(bp.maxSize)
-				logger.Infof("Backup %q: %.02f%% (%.02f/%.02fkb)", bp.filename, 100.0*written/maxSize, written/1024.0, maxSize/1024.0)
+				logger.Infof("%s %q: %.02f%% (%.02f/%.02fkb)",
+					bp.action,
+					bp.filename,
```
Comment (deepthi) [inline go/vt/mysqlctl/builtinbackupengine.go:711]: I suspect that for restore this still prints out the encoded filename (just a monotonically increasing number) and not the real filename. Did that change in your testing?

**Review by deepthi — COMMENTED** [review]
This is a good solution 👍 
The things that need to be fixed:
- DCO. If you don't already have the vitess git hooks installed, you will want to copy them from `misc/git/hooks` into your `.git` folder. `source build.env` does this for you as well
- This can't be easily unit tested because it is a log message. For such things we ask contributors to show the results from local testing. Can you post sample log from your local testing?
- You should create an issue and link it in the PR description with a `Fixes: ####`.

File: go/test/endtoend/backup/vtctlbackup/backup_utils.go [hunk 2]
```diff
@@ -998,7 +998,7 @@ func verifySemiSyncStatus(t *testing.T, vttablet *cluster.Vttablet, expectedStat
 }
 
 func terminateBackup(t *testing.T, alias string) {
-	stopBackupMsg := "Done taking Backup"
+	stopBackupMsg := "finished restore:"
```
Comment (deepthi) [inline go/test/endtoend/backup/vtctlbackup/backup_utils.go:1001]: I don't understand this. The test is calling `Backup` on line 1010. Why would we expect a `finished restore` message?

Comment (yields) [reply go/test/endtoend/backup/vtctlbackup/backup_utils.go]: Good catch, updated the log.

**Pull request closed without merging**